zcash_primitives/src/transaction/components/sapling/builder.rs

```diff
@@ -311,14 +311,7 @@ impl<P: consensus::Parameters> SaplingBuilder<P> {
         mut rng: R,
         target_height: BlockHeight,
         progress_notifier: Option<&Sender<Progress>>,
-    ) -> Result<
-        (
-            Vec<SpendDescription>,
-            Vec<OutputDescription>,
-            SaplingMetadata,
-        ),
-        Error,
-    > {
+    ) -> Result<(Option<Bundle<Unauthorized>>, SaplingMetadata), Error> {
```

Comment:
I think what we actually should return here is:
```suggestion
    ) -> Result<Option<(Bundle<Unauthorized>, SaplingMetadata)>, Error> {
```
since if there is no bundle, there is no metadata.

Reply:
This is currently an awkward change to make; I'm going to leave it for a future refactoring.